deploy/deploy-centralvenv.sh

```diff
@@ -0,0 +1,882 @@
+    echo "runFromSource: $runFromSource"
+    echo "noVenvCleanup: $noVenvCleanup"
+    echo "verboseMode: $verboseMode"
+    echo "central services host: $vmName"
+    echo "secSring: $secString"
+    echo "======================================================="
+    echo -n "Continue? [y]: "
+    read x && [[ $x =~ (n|N) ]] && return 102
+    echo "..."
+    echo "You still have 5 sec. to cancel before we proceed."
+    sleep 5
+}
+
+createVenv(){
+    # Cleanup your setup space
+    echo
+    echo "======================================================="
+    echo "Creating minimal virtual environment:"
+    echo -n "Continue? [y]: "
+    read x && [[ $x =~ (n|N) ]] && return 102
+    echo "..."
+
+    [[ -d $venvPath ]] || mkdir -p $venvPath || return $?
+    if $noVenvCleanup ; then
+        $pythonCmd -m venv $venvPath || return $?
+    else
+        $pythonCmd -m venv --clear $venvPath || return $?
+    fi
+}
+
+cloneWMCore(){
+    # clone deployment scripts
+    echo
+    echo "======================================================="
+    echo "Cloning WMCore source code:"
+    echo -n "Continue? [y]: "
+    read x && [[ $x =~ (n|N) ]] && return 101
+    echo "..."
+
+    [[ -d $wmSrcPath ]] ||  mkdir -p $wmSrcPath || return $?
+    cd $wmSrcPath
+    git clone $wmSrcRepo $wmSrcPath && git checkout $wmSrcBranch && [[ -n $wmTag ]] && git reset --hard $wmTag
+    cd -
+}
+
+setupConfig(){
+    # clone deployment scripts
+    echo
+    echo "======================================================="
+    echo "Cloning WMCore configuration files:"
+    echo -n "Continue? [y]: "
+    read x && [[ $x =~ (n|N) ]] && return 101
+    echo "..."
+
+    [[ -d $wmCfgPath ]] ||  mkdir -p $wmCfgPath || return $?
+    cd $wmCfgPath
+    git clone $wmCfgRepo $wmCfgPath
+    # git checkout $wmCfgBranch && git pull origin $wmCfgBranch || return $?
+
+    # First checkout to a fresh empty branch so we never mix with the origin branches:
+    git checkout -b currentConfig
+
+    for service in $enabledList
+    do
+        # Add enabled links:
+        echo "Touching: ${wmEnabledPath}/${service}"
+        touch ${wmEnabledPath}/${service} || { err=$? ; echo "Could not create enabled link for: $service"; return $err ;}
+
+        # Checkout needed service configs from the specified configuration branch:
+        echo "Cloning configuration for: $service"
+        git checkout origin/$wmCfgBranch -- $service || { err=$? ; echo "Could not checkout configuration for: $service from: origin/$wmCfgBranch"; return $err ;}
+    done
+    cd -
+}
+
+_pipUpgradeVenv(){
+    # upgrade pip for the current virtual env
+    cd $venvPath
+    # pip install $pipOpt wheel
+    # pip install $pipOpt --upgrade pip
+    pip install wheel
+    pip install --upgrade pip
+}
+
+activateVenv(){
+    # activate virtual environment
+    echo
+    echo "======================================================="
+    echo "Activate WMCore virtual env:"
+    echo -n "Continue? [y]: "
```

Comment:
once again, make sure it can run without user interactions

Reply:
done